enable/tools/apptools/commands.py

```diff
@@ -0,0 +1,236 @@
+#
+# (C) Copyright 2015 Enthought, Inc., Austin, TX
+# All right reserved.
+#
+# This file is open source software distributed according to the terms in
+# LICENSE.txt
+#
+
+"""
+Enable Commands
+===============
+
+This module provides :py:class:`apptools.undo.abstract_command.AbstractCommand`
+subclasses for common component manipulations, such as moving, resizing and
+setting attribute values.
+
+"""
+
+from __future__ import (division, absolute_import, print_function,
+                        unicode_literals)
+
+from apptools.undo.api import AbstractCommand
+from traits.api import Bool, Instance, Tuple, Unicode
+from traits.util.camel_case import camel_case_to_words
+
+from enable.component import Component
+
+class ComponentCommand(AbstractCommand):
+    """ Abstract command which operates on a Component """
+
+    #: The component the command is being performed on.
+    component = Instance(Component)
```

Comment:
I don't think it should we a weakref - if the `Component` we are referencing gets gc'd then we can't undo or redo the command.

Yes, this may mean that there are uncollected cycles, but as long as the command is on the command stack we want to keep everything around.  One of the problems with Undo/Redo is that you keep around stuff in memory that you would otherwise gc, because you need to remember it to be able to get back into a previous state.